torch/quantization/fx/quantize.py

```diff
@@ -491,6 +495,52 @@ def in_nodes(a: Argument, nodes: Set[Node]) -> bool:
 
     return result_graph
 
+def handle_cat_nodes(
+        model: torch.nn.Module, observed_graph: Graph, matches: Dict[str, MatchResult],
+        quants: Dict[str, List[Tuple[DefaultQuantizeHandler, Callable]]],
+        activation_post_process_map: Dict[str, List[str]],
+        modules: Dict[str, torch.nn.Module]):
+    observed_nodes: Set[Node] = set()
+    # activation_post_process for cat nodes
+    app_for_cat_nodes: Dict[Node, torch.nn.Module] = dict()
+
+    # we'll set the activation_post_process for all tensor inputs of cat and output
+    # of cat to be the same (the activation_post_process for the first Tensor
+    # input in the list, for example, if we have:
+    # x = torch.cat([x1, x2, x3], ...)
```

Comment:
is it

```
x1 -> obs1 -> cat -> obs1 -> ...
              /
      x2 -> obs1
```

or 

```
x1 -> list -> obs1 -> cat -> obs1 -> ...
        /
     x2
```

or something else?  Perhaps we can include the ascii of what's going on in the docblock?

Reply:
the first one, we don't have observers for list I think

Reply:
makes sense.  Could we add the graph to the docblock?  It would be really helpful for code readers.  

Seems like sharing observers for inputs of different sizes would bias the qparams towards the tensors with smaller sizes.  A workaround could be do weigh the contribution of different components of cat by the number of elements, but probably not worth it.  This PR is already much better compared to what we were doing before.